pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -13,25 +15,69 @@ import (
 	"vitess.io/vitess/go/vt/wrangler"
 )
 
+const (
+	maxSafeVReplicationLag = 10
+)
+
 func (r *ReconcileVitessKeyspace) reconcileResharding(ctx context.Context, vtk *planetscalev2.VitessKeyspace) error {
 	wr, ts, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
 	if err != nil {
 		return err
 	}
 	defer ts.Close()
 
-	workflows, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
+	workflowList, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
 	if err != nil {
-		workflows = make([]string, 0)
+		workflowList = make([]string, 0)
```

Comment:
We should bail out of `reconcileResharding` (with success but requeue, similar to topo failures) if we failed to list workflows, instead of claiming resharding is inactive.

We can borrow the code from VitessShard where we register status conditions in advance, so that they will be set to Unknown if we never give them an updated value.

Reply:
Done.